test/m3u8.test.js

```diff
@@ -108,6 +110,145 @@ QUnit.test('parses custom tags', function(assert) {
+  assert.ok(commentMapper2.called);
+  assert.strictEqual(dataCallback.callCount, 3);
+
+  assert.deepEqual(dataCallback.getCall(0).args[0], {
+    data: '#EXAMPLE',
+    type: 'custom',
+    customType: 'test',
+    segment: undefined
+  });
+  assert.deepEqual(dataCallback.getCall(1).args[0], {
+    data: '#NEW-COMMENT',
+    type: 'custom',
+    customType: 'test2',
+    segment: undefined
+  });
+  assert.deepEqual(dataCallback.getCall(2).args[0], {
+    text: 'SOMETHING-ELSE',
+    type: 'comment'
+  });
+});
+
+QUnit.test('maps custom tags', function(assert) {
+  const manifest = '#EXAMPLE\n';
+  const commentMapper = sinon.spy(line => '#NEW-COMMENT');
+  const dataCallback = sinon.spy();
+
+  this.parseStream.addTagMapper({
+    expression: /^#EXAMPLE/,
+    map: commentMapper
+  });
+
+  this.parseStream.on('data', dataCallback);
+  this.lineStream.push(manifest);
+
+  assert.ok(commentMapper.called);
+  assert.strictEqual(dataCallback.callCount, 2);
+
+  assert.deepEqual(dataCallback.getCall(0).args[0], {
+    text: 'EXAMPLE',
+    type: 'comment'
+  });
+  assert.deepEqual(dataCallback.getCall(1).args[0], {
+    text: 'NEW-COMMENT',
+    type: 'comment'
+  });
+});
+
+QUnit.test('maps multiple custom tags', function(assert) {
+  const manifest = '#VOD-STARTTIMESTAMP:1501533337573\n';
+  const commentMapper = sinon.spy(line => '#NEW-COMMENT');
+  const pdtMapper = sinon.spy((line) => {
+    const match = /#VOD-STARTTIMESTAMP:(\d+)/g.exec(line)[1];
+    const ISOdate = new Date(Number(match)).toISOString();
+
+    return `#EXT-X-PROGRAM-DATE-TIME:${ISOdate}`;
+  });
+  const dataCallback = sinon.spy();
+
+  this.parseStream.addTagMapper({
+    expression: /^#VOD-STARTTIMESTAMP/,
+    map: commentMapper
+  });
+  this.parseStream.addTagMapper({
+    expression: /^#VOD-STARTTIMESTAMP/,
+    map: pdtMapper
+  });
+
+  this.parseStream.on('data', dataCallback);
+  this.lineStream.push(manifest);
+
+  assert.ok(commentMapper.called);
+  assert.ok(pdtMapper.called);
+  assert.strictEqual(dataCallback.callCount, 3);
+
+  assert.deepEqual(dataCallback.getCall(0).args[0], {
+    text: 'VOD-STARTTIMESTAMP:1501533337573',
+    type: 'comment'
+  });
+
+  assert.deepEqual(dataCallback.getCall(1).args[0], {
+    text: 'NEW-COMMENT',
+    type: 'comment'
+  });
+
+  const dateTag = dataCallback.getCall(2).args[0];
+
+  assert.strictEqual(dateTag.dateTimeString, '2017-07-31T20:35:37.573Z');
+  assert.strictEqual(dateTag.tagType, 'program-date-time');
+  assert.strictEqual(dateTag.type, 'tag');
+});
+
+QUnit.test('mapper ignores tags', function(assert) {
+  const manifest = '#VOD-STARTTIMESTAMP:1501533337573\n';
+  let element;
+
+  this.parseStream.addTagMapper({
+    expression: /^#NO-MATCH/,
+    map(line) {
+      return '#MAPPED';
+    }
+  });
+
+  this.parseStream.on('data', function(elem) {
+    element = elem;
+  });
+
+  this.lineStream.push(manifest);
+  assert.strictEqual(element.type, 'comment', 'the type is comment');
```

Comment:
If you assert the number of data events here there are 2 instead of 1 (which I would expect if the line didn't need to be duplicated)